nRF Toolbox/Profiles/DeviceFirmwareUpdate/Views/FileScheme/FirmwareProgressImage.swift

```diff
@@ -0,0 +1,44 @@
+//
+//  FirmwareProgressImage.swift
+//  nRF Toolbox
+//
+//  Created by Nick Kibysh on 03/12/2019.
+//  Copyright © 2019 Nordic Semiconductor. All rights reserved.
+//
+
+import UIKit
+import iOSDFULibrary
+
+extension UIColor {
+    static let firmwareApplication: UIColor = .nordicRed
+    static let firmwareBootloader: UIColor = .nordicFall
+    static let firmwareSoftDevice: UIColor = .nordicGreen
+}
+
+class FirmwareProgressImage: ProgressImage {
+    func setParts(with firmware: DFUFirmware, reversed: Bool = false ) {
+        var parts = [ProgressPart]()
+        
+        let application = Int(firmware.size.application)
```

Comment:
In the current DFU lib version for packets with SD and BL (and optionally App) the BL size is set to 1 and the SD size is set to (SD+BL).size - 1.
I think you may have 2 parts: System (SD+BL) and App.

Reply:
done